File: iosApp/iosApp/Presentation/Main/CafeManagement/NoticeEvent/NoticeEventAction.swift
//
//  NoticeEventAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/11.
//

import Foundation

enum NoticeEventAction {
    case clickBack
    case selectTab(NoticeEventTab)
    case changeQuery(String)
    case loadMoreNotices
    case loadMoreEvents
    case clickRegister
    case clickMoreEvents
    case clickEditNotice(String)
    case clickDeleteNotice(String)
    case clickEventMenu(String)
    case dismissFormSheet
    case changeFormTitle(String)
    case changeFormContent(String)
    case clickFormImage
    case clickRemoveFormImage
    case changeFormPinned(Bool)
    case clickReserveSchedule
    case clickSubmitForm
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/noticeevent/NoticeEventAction.kt
package com.hhp227.concafe.presentation.main.cafemanagement.noticeevent

sealed interface NoticeEventAction {
    data object ClickBack : NoticeEventAction
    data class SelectTab(val tab: NoticeEventTab) : NoticeEventAction
    data class ChangeQuery(val value: String) : NoticeEventAction
    data object LoadMoreNotices : NoticeEventAction
    data object LoadMoreEvents : NoticeEventAction
    data object ClickRegister : NoticeEventAction
    data object ClickMoreEvents : NoticeEventAction
    data class ClickEditNotice(val id: String) : NoticeEventAction
    data class ClickDeleteNotice(val id: String) : NoticeEventAction
    data class ClickEditEvent(val id: String) : NoticeEventAction
    data class ClickDeleteEvent(val id: String) : NoticeEventAction
    data object DismissFormSheet : NoticeEventAction
    data class ChangeFormTitle(val value: String) : NoticeEventAction
    data class ChangeFormContent(val value: String) : NoticeEventAction
    data object ClickFormImage : NoticeEventAction
    data object ClickRemoveFormImage : NoticeEventAction
    data class ChangeFormPinned(val value: Boolean) : NoticeEventAction
    data object ClickReserveSchedule : NoticeEventAction
    data object ClickSubmitForm : NoticeEventAction
    data object DismissInfoMessage : NoticeEventAction
}
